### PR TITLE
Object factory changes

### PR DESCRIPTION
**Pull request by YuriySavchuk**

### CODE REVIEW

File: core/src/test/java/io/github/tcdl/impl/ResponderServerImplTest.java [hunk 1]
```diff
@@ -63,17 +64,17 @@ public void testResponderServerProcessSuccess() throws Exception {
...
     @Test
     public void testResponderServerProcessWithError() throws Exception {
         Exception error = new Exception();
-        ResponderServerImpl.RequestHandler handler = (request, responder) -> { throw error; };
+        ResponderServer.RequestHandler handler = (request, responder) -> { throw error; };
 
-        ResponderServerImpl responderServer = (ResponderServerImpl) ResponderServerImpl
-                .create(TestUtils.getSimpleNamespace(), messageTemplate, msbContext, handler)
-                .listen();
+        ResponderServerImpl responderServer = ResponderServerImpl
```
Comment (anstr) [inline core/src/test/java/io/github/tcdl/impl/ResponderServerImplTest.java:74]: Is it possible to use here ResponderServer responderServer = ResponderServerImpl ?

File: core/src/main/java/io/github/tcdl/MsbContextImpl.java [hunk 2]
```diff
@@ -100,71 +92,8 @@ public ObjectFactory getObjectFactory() {
...
-                });
-            }
-
-            ObjectFactoryImpl objectFactory = new ObjectFactoryImpl(msbContext); 
-            msbContext.setObjectFactory(objectFactory);
-            
-            return msbContext;
-        }
-    }
 }
```
Comment (rkatsyuryna) [inline core/src/main/java/io/github/tcdl/MsbContextImpl.java]: should we move this to io.github.tcdl.impl ? Then we can easy make our RequesterImpl, ResponderImpl and ResponderServerImpl create() methods package private and allow to create this only through our MsbContext

**Comment by rkatsyuryna** [comment]
would be nice to squash this two commits

File: core/src/test/java/io/github/tcdl/impl/ResponderServerImplTest.java (hunk 1 above)
Comment (rkatsyuryna) [inline core/src/test/java/io/github/tcdl/impl/ResponderServerImplTest.java]: no as we use here packege-private onResponder(). But one line above we can change:
 ResponderServerImpl.RequestHandler handler = (request, responder) -> { throw error; };
->
 ResponderServer.RequestHandler handler = (request, responder) -> { throw error; };

Comment (YuriySavchuk) [inline core/src/test/java/io/github/tcdl/impl/ResponderServerImplTest.java:74]: We really can't change type of responderServer variable. The test verifies functionality of ResponderServerImpl class. So some of the methods (onResponder() as an example) are accessible only via ResponderServerImpl variable.

**Pull request merged**